components/clp-py-utils/clp_py_utils/s3_utils.py

```diff
@@ -1,11 +1,137 @@
+import re
 from pathlib import Path
+from typing import List, Tuple
 
 import boto3
 from botocore.config import Config
 from botocore.exceptions import ClientError
+from job_orchestration.scheduler.job_config import S3InputConfig
 from result import Err, Ok, Result
 
 from clp_py_utils.clp_config import S3Config
+from clp_py_utils.compression import FileMetadata
+
+# Constants
+AWS_ENDPOINT = "amazonaws.com"
+
+
+def parse_aws_credentials_file(credentials_file_path: Path) -> Tuple[str, str]:
+    """
+    Parses the `aws_access_key_id` and `aws_secret_access_key` from the given credentials_file_path.
+    :param credentials_file_path: path to the file containing aws credentials.
+    :return: A tuple of (aws_access_key_id, aws_secret_access_key)
+    :raise: ValueError if the file doesn't exist, or doesn't contain the aws credentials.
+    """
+
+    aws_access_key_id = None
+    aws_secret_access_key = None
+
+    if not credentials_file_path.exists():
+        raise ValueError(f"File {credentials_file_path} doesn't exist.")
```

Comment:
```suggestion
        raise ValueError(f"'{credentials_file_path}' doesn't exist.")
```

I try to quote user inputs to make it obvious if they included a whitespace character by accident.